sktime/transformers/rocket.py

```diff
@@ -0,0 +1,178 @@
+from numba import njit, prange
+import numpy as np
+import pandas as pd
+
+from .base import BaseTransformer
+
+__author__ = "Angus Dempster"
+__all__ = ["ROCKET"]
+
+class Rocket(BaseTransformer):
+    """ROCKET
+
+    RandOm Convolutional KErnel Transform
+
+    @article{dempster_etal_2019,
+      author  = {Dempster, Angus and Petitjean, Fran\c{c}ois and Webb, Geoffrey I},
+      title   = {ROCKET: Exceptionally fast and accurate time series classification using random convolutional kernels},
+      year    = {2019},
+      journal = {arXiv:1910.13051}
+    }
+
+    Parameters
+    ----------
+    input_length : int, length of input time series
+    num_kernels  : int, number of random convolutional kernels
+    num_channels : int, number of channels for multivariate time series (default 1)
+    """
+
+    def __init__(self,  input_length, num_kernels, num_channels = 1):
+        self.input_length = input_length
+        self.num_kernels = num_kernels
+        self.num_channels = num_channels
+        self.kernels = _generate_kernels(input_length, num_kernels, num_channels)
+
+    def fit(self, X, y = None):
+        return self
+
+    def fit_transform(self, X, y = None):
+        return self.fit(X).transform(X)
```

Comment:
We can remove the `fit_transform` method, that comes automatically with the [`BaseTransformer`](https://github.com/scikit-learn/scikit-learn/blob/1495f6924/sklearn/base.py#L526).

Reply:
Do you mean removing `fit_transform` altogether from the base class (`BaseTransformer`)?

`fit` (and, by extension `fit_transform`) are irrelevant for ROCKET.

Reply:
Yes! Just to clarify in case it's not clear: the TransformerMixin form sklearn (and by extension our BaseTransformer) has a `fit_transform` method already which calls `.fit(X).transform(X)` which is inherited by your transformer, so no need to add (or overwrite) it here. Even if fit doesn't do anything, we should keep it here to have a consistent interface.